components/trustyai/trustyai.go

```diff
@@ -45,7 +47,7 @@ func (t *TrustyAI) OverrideManifests(ctx context.Context, _ cluster.Platform) er
 		if manifestConfig.SourcePath != "" {
 			defaultKustomizePath = manifestConfig.SourcePath
 		}
-		Path = filepath.Join(deploy.DefaultManifestPath, ComponentPathName, defaultKustomizePath)
+		OverridePath = filepath.Join(deploy.DefaultManifestPath, ComponentPathName, defaultKustomizePath)
```

Comment:
Having such a global state is not the best pattern. It even can bring problems with parallel execution.
Better approach would be if the function returns the path (well, as a shortcut "" if it was not overridden, but some extra boolean looks nicer). Then the logic below could be the same, "if overridden, replace the default".

Reply:
i am curious about this " _It even can bring problems with parallel execution_."
if you mean parallel as different components (trusty and kueue and so on) but the path variables are only defined within the package. or i missed something?

Reply:
> i am curious about this " _It even can bring problems with parallel execution_." if you mean parallel as different components (trusty and kueue and so on) but the path variables are only defined within the package. or i missed something?

Problems with params.env updating brings idea that we can encounter races when the same component is reconciled in parallel.

Reply:
back to the reason i kept using this way, not change it to 
`func (d *type) OverrideManifests(ctx context.Context, _ cluster.Platform) (string, error) {`
is to keep the scope of this change only for Trusty.

we can do the change in another PR, to get OverrideManifests re-write for other components as well.
It was the same reason, when i did the change for dashboard months ago, did not want to mix refactor code.
then, as usual, never occured to me afterwards :laughing: